docs/CreateKoblenzHash.md

```diff
@@ -0,0 +1,61 @@
+# Creating Hashes for Koblenz Pass Data
+
+## Steps
+
+The example data is
+
+| Parameter    | Value         |
+|--------------|---------------|
+| Name         | Karla Koblenz |
+| Geburtstag   | 10.06.2003    |
+| Aktenzeichen | 123K          |
+
+
+### 1. Collect all data and merge it into an object
+```agsl
+//Example:
+{
+    full_name: "Karla Koblenz"
+    birthday: 12213
+    referenceNumber: "123K"
+}
+```
+
+- The full_name must be `FirstnameSpaceLastname`. Every char must exactly match the user input, as otherwise there is not possibility to match the data Koblenz transfers with the input the user makes.
+e.g. `Karla Koblenz` will match neither with `Karla Lisa Koblenz` nor with `Karlá Koblenz`.
+- The birthday is defined in our protobuf [card.proto](../frontend/card.proto) file: It counts the days since the birthday (calculated from 1970-01-01).
+  All values of this field are valid, including the 0, which indicates that the birthday is on 1970-01-01. Birthdays before 1970-01-01 have negative values.
+- referenceNumber is set to the "Aktenzeichen"
+
+
+### 2. Convert this object to a Canonical Json
+ Result should be:
+ ```
+ {"1":"Karla Koblenz","2":12213,"3":"123K"}
+ ```
+
+### 3. Hash it with Argon2id
+
+Hash with Argon2id with the following parameters:
+
+| Parameter    | Value                                                                          |
+|--------------|--------------------------------------------------------------------------------|
+| Version      | 19                                                                             | 
+| Iterations   | 2                                                                              | 
+| Parallellism | 1                                                                              | 
+| Memory       | 19456 (19MiB)                                                                  | 
+| HashLength   | 32                                                                             | 
+| Salt         | Secret Salt will be shared with Koblenz<br/>for the example use `123456789ABC` | 
```

Comment:
Is it finally salt or not 🙈

Reply:
We use here a secret pepper shared with Koblenz as a additional security layer.

Reply:
I just meant that the terms are a bit mixed up, but maybe it doesn't matter

Reply:
i think the same terms should be used like in the online hash creator from argon